### ISSUE TITLE
right click menu and animation when an app is launched

### ISSUE THREAD

**Issue by DIDIK3V1** [spheras/desktopfolder#310]
Hello, I really like this app especially since my gnome desktop seems to not be working anymore, i would like you to add three things though. 
Firsly I'm on pop os and with this distro when right clicking on an app there is the ability to launch the app with the dedicated graphics card, but with this app there is no contextual right click menu on an app.
![image](https://user-images.githubusercontent.com/54941168/86589340-cedb6480-bf8d-11ea-9592-5521bb2d5a22.png)
Secondly when clicking on an app to launch there is no way to tell if the action has been considered by the app, adding an animation will help the user to understand that there is no need to keep clicking on the app
Finaly just add the option to open system setting with a right click on an empty space of the desktop.

**Comment by spheras** [comment]
Hi @DIDIK3V1, thanks for commenting on those issues. Regarding the first, it is difficult for me to support all the wide Linux ecosystem. Anyway, I'm really open if anyone is interested to offer a (generic) solution to it.

Regarding the others:

- It sounds interesting to me performing an animation when launching, I will try to keep it in mind for future releases.
- Opening system settings is also a good idea, why not. Not sure if there is a generic command for that.

**Comment by DIDIK3V1** [comment]
Okay in that case can you add the possibility to add arguments when launching an app on the right click menu so I can add it manually ? (Maybe doing the same thing for right click on the desktop, the ability to launch any app we want on a right click)

Envoyé depuis Yahoo Mail pour Android 
 
  Le lun., juil. 13, 2020 à 13:29, José Amuedo Salmerón<notifications@github.com> a écrit :   


Hi @DIDIK3V1, thanks for commenting on those issues. Regarding the first, it is difficult for me to support all the wide Linux ecosystem. Anyway, I'm really open if anyone is interested to offer a (generic) solution to it.

Regarding the others:
   
   - It sounds interesting to me performing an animation when launching, I will try to keep it in mind for future releases.
   - Opening system settings is also a good idea, why not. Not sure if there is a generic command for that.

**Comment by ferdnyc** [comment]
For 3, AFAIK there isn't a generic command, each desktop has its own settings tool. Unless maybe freedesktop standardized some D-Bus interface, but I don't think so. You'd probably have to find some way of identifying.

For the second request, it sounds like the app that's being launched doesn't support [startup notification](https://developer.gnome.org/integration-guide/stable/startup-notification.html.en) (or the desktop environment doesn't), normally that's how an app would communicate that it's in the process of opening. The launcher / file manager tool shouldn't have to be responsible for that. (I mean... sure, it _can_, but there's no way to do it in a manner that's consistent with the rest of the system.)

**Comment by DIDIK3V1** [comment]
Yeah that's why you can just add the option to choose the app to launch ;) And for the animation it is just to signify that the instruction has been launched so just an animation when clicked on the app the right way (sometimes when I double click I accidentally grap the appicon so for exemple this will signify the use the command has been launched)
